allenact/algorithms/onpolicy_sync/runner.py

```diff
@@ -630,50 +655,77 @@ def experiment_name(self):
     def checkpoint_dir(
         self, start_time_str: Optional[str] = None, create_if_none: bool = True
     ):
-        folder = os.path.join(
-            self.output_dir,
-            "checkpoints",
+        path_parts = [
             self.config.tag()
             if self.extra_tag == ""
             else os.path.join(self.config.tag(), self.extra_tag),
             start_time_str or self.local_start_time_str,
-        )
+        ]
+        if self.save_dir_fmt == SaveDirFormat.NESTED.value:
+            folder = os.path.join(self.output_dir, *path_parts, "checkpoints",)
+        elif self.save_dir_fmt == SaveDirFormat.FLAT.value:
+            folder = os.path.join(self.output_dir, "checkpoints", *path_parts,)
         if create_if_none:
             os.makedirs(folder, exist_ok=True)
         return folder
 
-    def log_writer_path(self, start_time_str: str) -> str:
-        path = os.path.join(
-            self.output_dir,
-            "tb",
-            self.config.tag()
-            if self.extra_tag == ""
-            else os.path.join(self.config.tag(), self.extra_tag),
-            start_time_str,
-        )
-        if self.mode == TEST_MODE_STR:
-            path = os.path.join(path, "test", self.local_start_time_str)
-        return path
-
-    def metric_path(self, start_time_str: str) -> str:
-        return os.path.join(
-            self.output_dir,
-            "metrics",
-            self.config.tag()
-            if self.extra_tag == ""
-            else os.path.join(self.config.tag(), self.extra_tag),
-            start_time_str,
-        )
+    def log_writer_path(self, start_time_str: str, checkpoint_parts: List = []) -> str:
```

Comment:
This function will be simpler when we infer and set the output dir earlier on rather than using the `checkpoint_parts`.